### ISSUE TITLE
Request ability to enable Istio support

### ISSUE THREAD

**Issue by hmklabs-dclow** [emissary-ingress/emissary#3293]
Please describe your use case / problem.
As a user of Ambassador and Istio I would like to be able to deploy Ambassador with Istio support enabled using the Ambassador Helm chart.

Describe the solution you'd like
It would be helpful if there was a configuration option in the Helm chart that would add the configuration for the Istio sidecar into the Ambassador installation.

Describe alternatives you've considered
There is a recent document on the steps to enable Istio support. However, this would need to be configured outside the Helm chart.

https://www.getambassador.io/docs/latest/howtos/istio/

**Comment by screamyao** [comment]
I stumbled upon @nicolasbelanger's solution above and had to make a couple more tweaks to make it work for us. 

1. Changed the default user of Ambassador to match Istio
2. Added `AMBASSADOR_ISTIO_SECRET_DIR` env variable

Version:
**Istio**: 1.7.0
**Ambassador**: 1.12.2

Changes to `values.yaml`:

```yaml
env: 
  AMBASSADOR_ISTIO_SECRET_DIR: "/etc/istio-certs"
  AMBASSADOR_ENVOY_BASE_ID: "1"

security:
  podSecurityContext:
    runAsUser: 1337  # Default was 8888 while Istio was 1337. Ambassador was unable to read from the volume as user 8888

podAnnotations:
  proxy.istio.io/config: |
    proxyMetadata:
      OUTPUT_CERTS: /etc/istio-certs
  sidecar.istio.io/userVolume: '[{"name": "istio-certs", "emptyDir": {"medium": "Memory"}}]'
  sidecar.istio.io/userVolumeMount: '[{"name": "istio-certs", "mountPath": "/etc/istio-certs"}]'
  traffic.sidecar.istio.io/includeInboundPorts: ''
  traffic.sidecar.istio.io/includeOutboundIPRanges: ''

volumes:
  - name: istio-certs
    emptyDir:
      medium: Memory

volumeMounts:
  - mountPath: /etc/istio-certs/
    name: istio-certs
```

**Comment by ysaakpr** [comment]
@screamyao  do I need to add istio-proxy side car also the the values override for istio integration with emissary? 
As mentioned here https://www.getambassador.io/docs/emissary/latest/howtos/istio/#istio-mutual-tls

**Comment by cindymullins-dw** [comment]
The ability to deploy Ambassador with Istio support enabled, using the Ambassador Helm chart, is available with version 2.2.
https://www.getambassador.io/docs/edge-stack/latest/howtos/istio/#installation-with-helm-recommended (@hmklabs-dclow, @nicolasbelanger , @kmattiasek1, @screamyao)

@ysaakpr, the Istio-proxy sidecar should be in the service so you would need to add that.
